caql/CaQLLexer.g4

```diff
@@ -0,0 +1,103 @@
+lexer grammar CaQLLexer;
+
+fragment NUMERAL: [0-9]+ ('.' [0-9]+)?;
+
+fragment SCIENTIFIC_NUMBER
+   : NUMERAL ('e' ('-' | '+')? NUMERAL)?
+   ;
+
+NUMBER
+    : NUMERAL
+    | SCIENTIFIC_NUMBER;
+STRING
+    : '\'' (~('\'' | '\\') | '\\' .)* '\''
+    | '"' (~('"' | '\\') | '\\' .)* '"'
+    ;
+
+// Binary operators
+
+ADD:  '+';
+SUB:  '-';
+MULT: '*';
+DIV:  '/';
+MOD:  '%';
+POW:  '^';
+
+AND:    'and';
+OR:     'or';
+UNLESS: 'unless';
+
+// Comparison operators
+
+EQ:  '=';
+DEQ: '==';
+NE:  '!=';
+GT:  '>';
+LT:  '<';
+GE:  '>=';
+LE:  '<=';
+RE:  '=~';
+NRE: '!~';
+
+// Aggregation modifiers
+
+BY:      'by';
+WITHOUT: 'without';
+
+// Join modifiers
+
+ON:          'on';
+IGNORING:    'ignoring';
+GROUP_LEFT:  'group_left';
+GROUP_RIGHT: 'group_right';
+
+OFFSET: 'offset';
+
+BOOL: 'bool';
+AGGREGATION_OPERATOR
+    : 'stats:sum'
+    | 'label'
+    | 'top'
+    | 'count'
+    | 'stats:max'
+    | 'stats:percentile'
+    | 'histogram'
+    | 'histogram:percentile'
+    ;
+
+FUNCTION
+    : 'find'
+    | 'find:counter'
+    | 'find:average'
+    | 'op:prod'
+    | 'op:div'
+    ;
+
+LEFT_BRACE:  '{';
+RIGHT_BRACE: '}';
+
+LEFT_PAREN:  '(';
+RIGHT_PAREN: ')';
+
+LEFT_BRACKET:  '[';
+RIGHT_BRACKET: ']';
+
+COMMA: ',';
+
+Line: '|';

```

Comment:
The CSharp target of Antlr4 doesn't like `Line` (`CaQLLexer.cs(41,13): warning CS0108: 'CaQLLexer.Line' hides in
herited member 'Lexer.Line'. Use the new keyword if hiding was intended.`). You should rename `Line` to something else, like `Line_` or `VBAR`.